pkg/vault/test/foundry/RecoveryMode.t.sol

```diff
@@ -49,6 +51,39 @@ contract RecoveryModeTest is BaseVaultTest {
         router.removeLiquidityRecovery(pool, amountToRemove);
     }
 
+    function testRecoveryModeWithRoundtripFee() public {
+        // Add initial liquidity.
+        uint256[] memory amountsIn = [uint256(defaultAmount), uint256(defaultAmount)].toMemoryArray();
+
+        vm.prank(alice);
+        (, uint256 bptAmountOut, ) = router.addLiquidityCustom(pool, amountsIn, bptAmount, false, bytes(""));
+
+        // Put pool in recovery mode.
+        vault.manualEnableRecoveryMode(pool);
+        // Set the fee, and flag to trigger collection of it.
+        vault.manualSetStaticSwapFeePercentage(pool, BASE_MAX_SWAP_FEE);
+        vault.manualSetAddLiquidityCalledFlag(pool, true);
```

Comment:
You don't need this one actually. The whole test runs in the same tx.

Maybe just leave a comment, and / or assert that the flag is true?

Reply:
I did this, but it doesn't seem to be testing the content of the events properly. If that flag is set, it should be charging fees in the original test too, but the expected event doesn't have them, yet it passes. I manually clear the flag in the one that's not supposed to have the roundtrip fee, and that works too.

Testing the balances verifies that the two cases behave differently, but I wonder where else events aren't getting tested.